clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala

```diff
@@ -78,6 +83,43 @@ trait PersistenceDAO extends LazyLogging {
       )
       logger.debug(s"Wrote document $document to ${written.toUri}")
     }
+
+  /**
+    * Return the GCS bucket metadata files for `index` upserted on or
+    * after `upsertId`.  Decode the files from JSON as `ClioDocument`s
+    * sorted by `upsertId`.
+    *
+    * @param upsertId of the last known upserted document
+    * @param index  to query against.
+    * @param ec is the implicit ExecutionContext
+    * @tparam D is a ClioDocument type with a JSON Decoder
+    * @return a Future Seq of ClioDocument
+    */
+  def getAllSince[D <: ClioDocument: Decoder](
+    upsertId: UUID,
+    index: ElasticsearchIndex[D]
+  )(implicit ec: ExecutionContext): Future[Seq[D]] =
+    Future {
+      val rootDirectory = rootPath.resolve(index.currentPersistenceDir)
+      val files = Try(Files.newDirectoryStream(rootDirectory))
+        .map(_.asScala)
+        .fold(throw _, identity)
+        .toSeq
+      val filesWithId = files.filter(_.endsWith(s"${upsertId}.json"))
+      val withId = filesWithId match {
+        case Seq(result) => result
+        case _ =>
+          throw new RuntimeException(
+            s"${filesWithId.size} files end with ${upsertId.toString}.json"
+          )
+      }
+      val (_, after) = files.span(path => !path.equals(withId))
```

Comment:
Not sure if it matters, but should we worry about the performance of taking multiple passes over the whole `Seq` of filenames?

Reply:
I'll worry when we discover a problem.

We could eliminate one pass and shorten another if we had the path of the first file to re-upsert.